ValheimPerformanceOptimizations/ZDOManPatches.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using HarmonyLib;
+using UnityEngine;
+
+namespace ValheimPerformanceOptimizations
+{
+    /// <summary>
+    /// Loading big worlds is slow and can partially be called threaded to speed it up.
+    /// </summary>
+    [HarmonyPatch(typeof(ZDOMan), "Load")]
+    public static class ZDOManPatches
+    {
+        private static bool Prefix(ZDOMan __instance, BinaryReader reader, int version)
+        {
+            // only patch the current data version to not risk any world file breaking
+            if (version != 27)
+            {
+                string message = $"ZDOMan.Load() unsupported data version: {version}. Fallback to original method";
+                ValheimPerformanceOptimizations.Logger.LogInfo(message);
+                return true;
+            }
+
+            reader.ReadInt64();
+            uint num = reader.ReadUInt32();
+            int num2 = reader.ReadInt32();
+            ZDOPool.Release(__instance.m_objectsByID);
+            __instance.m_objectsByID.Clear();
+            __instance.ResetSectorArray();
+            ZLog.Log("Loading " + num2 + " zdos , my id " + __instance.m_myid + " data version:" + version);
+
+            /* --- begin changes --- */
+
+            Tuple<ZDO, byte[]>[] packages = new Tuple<ZDO, byte[]>[num2];
+
+            // don't load zDO data immediately
+            for (int i = 0; i < num2; i++)
+            {
+                ZDO zDO = ZDOPool.Create(__instance);
+                zDO.m_uid = new ZDOID(reader);
+                int count = reader.ReadInt32();
+                byte[] data = reader.ReadBytes(count);
+                __instance.m_objectsByID.Add(zDO.m_uid, zDO);
+
+                // store the data
+                packages[i] = new Tuple<ZDO, byte[]>(zDO, data);
+
+                if (zDO.m_uid.userID == __instance.m_myid && zDO.m_uid.id >= num)
+                {
+                    num = zDO.m_uid.id + 1;
+                }
+            }
+
+            // now load zDO threaded
+            // batch size of 2^12 - 2^16 works good
+            const int batch = 32768;
```

Comment:
var batch = Math.Min(num2, 32768)

Reply:
I think the acutual problem is with `num2 / batch` because it is implicit casted to an int and therefore one iteration is missing